src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts

```diff
@@ -0,0 +1,119 @@
+import { RequiredBuilder } from '@/app/collective-rewards/types'
+import { useGaugesGetFunction } from '@/app/collective-rewards/shared'
+import { useCycleContext } from '@/app/collective-rewards/metrics'
+import { formatEther } from 'viem'
+import { useMemo } from 'react'
+
+export const useGetABI = () => {
+  const {
+    data: builders,
+    isLoading: buildersLoading,
+    error: buildersError,
+  } = useGetBuildersByState<RequiredBuilder>({
+    activated: true,
+    communityApproved: true,
+    kycApproved: true,
+    revoked: false,
+  })
+
+  const { data: rifRewards, isLoading: rifRewardsLoading, error: rifRewardsError } = useGetRewardsERC20()
+  const {
+    data: rbtcRewards,
+    isLoading: rbtcRewardsLoading,
+    error: rbtcRewardsError,
+  } = useGetRewardsCoinbase()
+
+  const {
+    data: { cycleNext },
+    isLoading: cycleLoading,
+    error: cycleError,
+  } = useCycleContext()
+
+  const gauges = builders.map(({ gauge }) => gauge)
+  const {
+    data: totalAllocation,
+    isLoading: totalAllocationLoading,
+    error: totalAllocationError,
+  } = useGaugesGetFunction(gauges, 'totalAllocation')
+
+  const buildersAddress = builders.map(({ address }) => address)
+  const {
+    data: backersRewardsPct,
+    isLoading: backersRewardsPctLoading,
+    error: backersRewardsPctError,
+  } = useGetBackersRewardPercentage(buildersAddress, cycleNext.toSeconds())
+
+  const { prices } = usePricesContext()
+
+  const abi = useMemo(() => {
+    const sumTotalAllocation = Object.values(totalAllocation).reduce((acc, value) => acc + (value ?? 0n), 0n)
+
+    if (!sumTotalAllocation) {
+      return 0
+    }
+
+    const rifPrice = prices.RIF?.price ?? 0
+    const rbtcPrice = prices.RBTC?.price ?? 0
+    const rifAmount = Number(formatEther(rifRewards ?? 0n))
+    const rbtcAmount = Number(formatEther(rbtcRewards ?? 0n))
+    const cyclePayout = rifAmount * rifPrice + rbtcAmount * rbtcPrice
+
+    if (!rifPrice) {
+      return 0
+    }
+
+    const topFiveBuilders = builders
+      .reduce<Array<{ allocation: bigint; current: number }>>((acc, builder) => {
+        const allocation = totalAllocation[builder.gauge]
+        const rewardPct = backersRewardsPct[builder.address]
+        if (allocation && rewardPct) {
+          acc.push({ allocation, current: rewardPct.current })
+        }
+        return acc
+      }, [])
+      .sort((a, b) => (a.allocation > b.allocation ? -1 : 1))
+      .slice(0, 5)
+
+    const weightedAverageBuilderRewardsPct =
+      topFiveBuilders.reduce(
+        (acc, { allocation, current }) =>
+          acc + Number(((allocation * 100n) / sumTotalAllocation) * BigInt(current)),
+        0,
+      ) / 100
+
+    const totalAllocationInEther = Number(formatEther(sumTotalAllocation))
+    const rewardsPerStRIFPerCycle =
+      (cyclePayout * (weightedAverageBuilderRewardsPct / totalAllocationInEther / rifPrice)) / 100
+
+    return (Math.pow(1 + rewardsPerStRIFPerCycle, 26) - 1) * 100
```

Comment:
@franciscotobar why do we need to divide by 100 and then multiply by 100? I think the formula provided was:
```suggestion
    const rewardsPerStRIFPerCycle = cyclePayout * (weightedAverageBuilderRewardsPct / totalAllocationInEther)
    return Math.pow(1 + rewardsPerStRIFPerCycle / rifPrice, 26) - 1
```